python/lsst/daf/relation/_operations/_deduplication.py

```diff
@@ -53,7 +53,13 @@ def __str__(self) -> str:
 
     def applied_min_rows(self, target: Relation) -> int:
         # Docstring inherited.
-        return 1
+        return 1 if target.min_rows >= 1 else 0
+
+    def applied_max_rows(self, target: Relation) -> int | None:
+        # Docstring inherited.
+        if not target.columns:
+            return 1 if target.max_rows is None or target.max_rows >= 1 else 0
```

Comment:
Out of curiosity why can this ever be '1' instead of '0'?  Doesn't no columns also imply no rows?

Reply:
Not quite; if you think of it as a `list` of `dict` rows it's the difference between `[]` and `[{}]`.  But where it really matters is actually in what happens when you join a relation like this to a regular one with nonzero numbers of rows and columns: `[{}]` is the "join identity" - a join between it and any other relation is the other relation, while `[]` is the "join zero" relation - a join between it and any other relation yields a relation with no rows (but the other relation's columns).